IllegalArgumentException from NetworkUtils

I've seen a couple of these in the play store now.

If I had to guess media.getDownload_url() is returning null or an empty string.

Kind of hard to tell because we're not getting much help from the stack trace.  Perhaps adding the onError handler would help?

```
java.lang.IllegalStateException: Exception thrown on Scheduler.Worker thread. Add `onError` handling.
at rx.internal.schedulers.ScheduledAction.run(ScheduledAction.java:60)
at android.os.Handler.handleCallback(Handler.java:615)
at android.os.Handler.dispatchMessage(Handler.java:92)
at android.os.Looper.loop(Looper.java:137)
at android.app.ActivityThread.main(ActivityThread.java:4898)
at java.lang.reflect.Method.invokeNative(Native Method)
at java.lang.reflect.Method.invoke(Method.java:511)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1006)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:773)
at dalvik.system.NativeStart.main(Native Method)
Caused by: rx.exceptions.OnErrorNotImplementedException: unexpected url: 
at rx.Observable$30.onError(Observable.java:7540)
at rx.observers.SafeSubscriber._onError(SafeSubscriber.java:154)
at rx.observers.SafeSubscriber.onError(SafeSubscriber.java:111)
at rx.internal.operators.OperatorObserveOn$ObserveOnSubscriber.pollQueue(OperatorObserveOn.java:197)
at rx.internal.operators.OperatorObserveOn$ObserveOnSubscriber$2.call(OperatorObserveOn.java:170)
at rx.internal.schedulers.ScheduledAction.run(ScheduledAction.java:55)
... 9 more
Caused by: java.lang.IllegalArgumentException: unexpected url: 
at com.squareup.okhttp.Request$Builder.url(Request.java:157)
at de.danoeh.antennapod.core.util.NetworkUtils$1.call(NetworkUtils.java:1113)
at rx.Observable.unsafeSubscribe(Observable.java:7713)
at rx.internal.operators.OperatorSubscribeOn$1$1.call(OperatorSubscribeOn.java:62)
at rx.internal.schedulers.ScheduledAction.run(ScheduledAction.java:55)
at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:442)
at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:305)
at java.util.concurrent.FutureTask.run(FutureTask.java:137)
at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:150)
at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:264)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1076)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:569)
at java.lang.Thread.run(Thread.java:856)
```

--- a/app/src/main/java/de/danoeh/antennapod/adapter/AdapterUtils.java
+++ b/app/src/main/java/de/danoeh/antennapod/adapter/AdapterUtils.java
@@ -1,5 +1,6 @@
 package de.danoeh.antennapod.adapter;
 
+import android.util.Log;
 import android.view.View;
 import android.widget.ProgressBar;
 import android.widget.TextView;
@@ -52,13 +53,16 @@ public class AdapterUtils {
                 txtvPos.setText("{fa-spinner}");
                 Iconify.addIcons(txtvPos);
                 NetworkUtils.getFeedMediaSizeObservable(media)
-                        .subscribe(size -> {
+                        .subscribe(
+                                size -> {
                                     if (size > 0) {
                                         txtvPos.setText(Converter.byteToString(size));
                                     } else {
                                         txtvPos.setText("");
                                     }
-                        });
+                                }, error -> {
+                                    Log.e(TAG, Log.getStackTraceString(error));
+                                });
             } else {
                 txtvPos.setText("");
             }

--- a/core/src/main/java/de/danoeh/antennapod/core/util/NetworkUtils.java
+++ b/core/src/main/java/de/danoeh/antennapod/core/util/NetworkUtils.java
@@ -5,6 +5,7 @@ import android.net.ConnectivityManager;
 import android.net.NetworkInfo;
 import android.net.wifi.WifiInfo;
 import android.net.wifi.WifiManager;
+import android.text.TextUtils;
 import android.util.Log;
 
 import com.squareup.okhttp.OkHttpClient;
@@ -108,9 +109,17 @@ public class NetworkUtils {
                     }
                 } else if (false == media.checkedOnSizeButUnknown()) {
                     // only query the network if we haven't already checked
+
+                    String url = media.getDownload_url();
+                    if(TextUtils.isEmpty(url)) {
+                        subscriber.onNext(0L);
+                        subscriber.onCompleted();
+                        return;
+                    }
+
                     OkHttpClient client = AntennapodHttpClient.getHttpClient();
                     Request.Builder httpReq = new Request.Builder()
-                            .url(media.getDownload_url())
+                            .url(url)
                             .header("Accept-Encoding", "identity")
                             .head();
                     try {

